### DISCUSSION TITLE
What does this error mean - Raspberry PI 4

### DISCUSSION THREAD

**Discussion by brianreinhold** [rm5248/JavaSerial#32] [Q&A]
I get this error when I call the method SerialPort.getSerialPorts()

ERROR: This is a very bad thing that should never happen /home/runner/work/JavaSerial/JavaSerial/main/NativeCode/SerialPortImpl.c:1952  errno:6

Its just a 'No such device or address' BUT it causes the system to crash and reboot  - sometimes immediately and sometimes not until after about a 10-20 seconds.  Looking at the C code where this error occurs it is not clear to me why this should cause a crash unless the errors are not handled at all.

Never see this error on Windows 10

I see it on Raspberry PI 3 versions too.

**Comment by rm5248** [comment]
Errno 6 is apparently 'no such device or address.'  I would be surprised if this caused the system to reboot, since all it is doing is attempting to open the device to see if it is a tty.   Simply attempting to open a device should not cause the system to reboot.

**Reply by brianreinhold** [reply]
Yea, I don't get it. I have noted when I run the app as root the error is much more likely to cause a reboot. The weird thing is that I get the error EVEN when the SerialPort.getSerialPorts() returns a non-empty list (for example I have a cable connected. As root, the reboot still occurs. I don't know how to debug it.

**Reply by rm5248** [reply]
You could try using `cat` to read something from every file, see if that causes a reboot.
